tests/ordering/order_test.go

```diff
@@ -241,6 +256,55 @@ func TestQCFutureOrdering(t *testing.T) {
 	}
 }
 
+func TestAsyncHandlerOrdering(t *testing.T) {
+	defer leakcheck.Check(t)
+	// servers setup
+	numServers := 4
+	srvs := make([]*GorumsServer, numServers)
+	addrs := make([]string, numServers)
+	for i := 0; i < numServers; i++ {
+		srv := NewGorumsServer()
+		srv.RegisterGorumsTestServer(&testSrv{})
+		lis, err := getListener()
+		if err != nil {
+			t.Fatalf("Failed to listen on port: %v", err)
+		}
+		addrs[i] = lis.Addr().String()
+		srvs[i] = srv
+		go srv.Serve(lis)
+	}
+
+	// client setup
+	man, err := NewManager(addrs,
+		WithGrpcDialOptions(grpc.WithInsecure()),
+		WithDialTimeout(10*time.Second),
+	)
+	if err != nil {
+		t.Fatalf("Failed to create manager: %v", err)
+	}
+	c, err := man.NewConfiguration(man.NodeIDs(), &testQSpec{})
+	if err != nil {
+		t.Fatalf("Failed to create configuration: %v", err)
+	}
+
+	// begin test
+	numRuns := 1 << 16
+	for i := 1; i < numRuns; i++ {
+		resp, err := c.QC(context.Background(), &Request{Num: uint64(i)})
```

Comment:
Doesn't seem to be calling the `AsyncHandler()` RPC.

Reply:
I should probably think while using copy-paste :facepalm: